_studio/shared/umc/codec/h264_dec/src/umc_h264_dec_bitstream_headers.cpp

```diff
@@ -335,6 +335,44 @@ H264HeadersBitstream::H264HeadersBitstream(uint8_t * const pb, const uint32_t ma
 {
 }
 
+//return ture if level_idc is correct
+inline bool H264HeadersBitstream::CheckLevel(uint8_t profile_idc, uint8_t level_idc)
+{
+    switch(level_idc)
+    {
+    case H264VideoDecoderParams::H264_LEVEL_1:
+    case H264VideoDecoderParams::H264_LEVEL_11:
+    case H264VideoDecoderParams::H264_LEVEL_12:
+    case H264VideoDecoderParams::H264_LEVEL_13:
+
+    case H264VideoDecoderParams::H264_LEVEL_2:
+    case H264VideoDecoderParams::H264_LEVEL_21:
+    case H264VideoDecoderParams::H264_LEVEL_22:
+
+    case H264VideoDecoderParams::H264_LEVEL_3:
+    case H264VideoDecoderParams::H264_LEVEL_31:
+    case H264VideoDecoderParams::H264_LEVEL_32:
+
+    case H264VideoDecoderParams::H264_LEVEL_4:
+    case H264VideoDecoderParams::H264_LEVEL_41:
+    case H264VideoDecoderParams::H264_LEVEL_42:
+
+    case H264VideoDecoderParams::H264_LEVEL_5:
+    case H264VideoDecoderParams::H264_LEVEL_51:
+    case H264VideoDecoderParams::H264_LEVEL_52:
+        return true;
+    case H264VideoDecoderParams::H264_LEVEL_9:
+        if (profile_idc != H264VideoDecoderParams::H264_PROFILE_BASELINE &&
+            profile_idc != H264VideoDecoderParams::H264_PROFILE_MAIN &&
+            profile_idc != H264VideoDecoderParams::H264_PROFILE_EXTENDED) {
+                level_idc = H264VideoDecoderParams::H264_LEVEL_1b;
```

Comment:
no sense in this assignment, since ```level_idc``` is not a reference. Anyway, should this function correct value of ```level_idc``` somehow? If so, the name is missleading

Reply:
@aulitin, please leave `sps->level_idc` adjustment at its prev. place - at `GetSequenceParamSet`